src/linker/Linker.Dataflow/ReflectionMethodBodyScanner.cs

```diff
@@ -1571,13 +1573,23 @@ void RequireDynamicallyAccessedMembers (ref ReflectionPatternContext reflectionC
 				} else if (uniqueValue is SystemTypeValue systemTypeValue) {
 					MarkTypeForDynamicallyAccessedMembers (ref reflectionContext, systemTypeValue.TypeRepresented, requiredMemberTypes);
 				} else if (uniqueValue is KnownStringValue knownStringValue) {
-					TypeDefinition foundType = _context.TypeNameResolver.ResolveTypeName (knownStringValue.Contents)?.Resolve ();
+					TypeReference typeRef = _context.TypeNameResolver.ResolveTypeName (knownStringValue.Contents);
+					TypeDefinition foundType = typeRef?.Resolve ();
 					if (foundType == null) {
 						// Intentionally ignore - it's not wrong for code to call Type.GetType on non-existing name, the code might expect null/exception back.
 						reflectionContext.RecordHandledPattern ();
 					} else {
+						MarkType (ref reflectionContext, typeRef);
 						MarkTypeForDynamicallyAccessedMembers (ref reflectionContext, foundType, requiredMemberTypes);
 					}
+				} else if (uniqueValue is ArrayValue arrayValue) {
+					TypeReference typeReference = _context.TypeNameResolver.ResolveTypeName ("System.Array");
+					TypeDefinition systemArrayType = typeReference?.Resolve ();
+					if (systemArrayType == null) {
+						reflectionContext.RecordHandledPattern ();
+					} else {
+						MarkTypeForDynamicallyAccessedMembers (ref reflectionContext, systemArrayType, requiredMemberTypes);
+					}
```

Comment:
This should never happen - it would mean that the application code is using array in place of Type. Or that annotations are completely wrong. This should be removed.